lib/diagnostic.rb

```diff
@@ -43,6 +43,6 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.readlines(filename).chomp.join(' ')
```

Comment:
This may work, but the instructions as for the File.open tactic

```ruby
File.open('filename') do |file|
  file.each do |line|
   line.chomp.join(' ' )
  end
end
```

something along these lines